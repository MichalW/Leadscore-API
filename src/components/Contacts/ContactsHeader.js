import React from 'react';
import {Table} from 'semantic-ui-react';

const ContactsHeader = () => (
  <Table.Header>
    <Table.Row>
      <Table.HeaderCell>First name</Table.HeaderCell>
      <Table.HeaderCell>Last name</Table.HeaderCell>
      <Table.HeaderCell>Email</Table.HeaderCell>
    </Table.Row>
  </Table.Header>
);

export default ContactsHeader;
