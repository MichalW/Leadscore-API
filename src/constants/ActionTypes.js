export const SIGN_IN_REQUEST = 'SIGN_IN_REQUEST';
export const SIGN_IN_SUCCESS = 'SIGN_IN_SUCCESS';
export const SIGN_IN_ERROR = 'SIGN_IN_ERROR';

export const SIGN_OUT_REQUEST = 'SIGN_OUT_REQUEST';
export const SIGN_OUT_SUCCESS = 'SIGN_OUT_SUCCESS';
export const SIGN_OUT_ERROR = 'SIGN_OUT_ERROR';

export const CONTACTS_FETCH_REQUEST = 'CONTACTS_FETCH_REQUEST';
export const CONTACTS_FETCH_SUCCESS = 'CONTACTS_FETCH_SUCCESS';
export const CONTACTS_FETCH_ERROR = 'CONTACTS_FETCH_ERROR';

export const ERROR = 'ERROR';
