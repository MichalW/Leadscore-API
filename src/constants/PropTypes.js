import {
  arrayOf,
  func,
  number,
  shape,
  string,
} from 'prop-types';

export const emailType = shape({
  email: string,
});

export const contactType = shape({
  created: number,
  emails: arrayOf(emailType),
  firstName: string,
  lastName: string,
  profilePicture: string,
});

export const historyType = shape({
  push: func.isRequired,
});

export const locationType = shape({
  pathname: string,
});

export const matchType = shape({
  params: shape({}),
});
