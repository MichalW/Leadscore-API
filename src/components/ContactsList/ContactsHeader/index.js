import {withRouter} from 'react-router-dom';

import ContactsHeader from './ContactsHeader';

export default withRouter(ContactsHeader);
