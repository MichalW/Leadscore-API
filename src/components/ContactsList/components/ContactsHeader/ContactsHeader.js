import React, {PureComponent} from 'react';
import {Table} from 'semantic-ui-react';

import {
  ASC, ASCENDING, DESC, DESCENDING,
} from '../../../../constants/ActionTypes';
import {DEFAULT_CONTACTS_SORT, DEFAULT_CONTACTS_SORT_DIRECTION} from '../../../../config';
import {historyType, matchType} from '../../../../constants/PropTypes';

const tableDirectionMap = {
  ASC: ASCENDING,
  DESC: DESCENDING,
};

class ContactsHeader extends PureComponent {
  constructor(props) {
    super(props);
    const {match: {params: {column, direction}}} = props;

    this.state = {
      column: column || DEFAULT_CONTACTS_SORT,
      direction: direction || DEFAULT_CONTACTS_SORT_DIRECTION,
    };
  }

  handleSort = clickedColumn => () => {
    const {column, direction} = this.state;

    if (column !== clickedColumn) {
      this.setState({
        column: clickedColumn,
        direction: ASC,
      }, this.historyPush);
    } else {
      this.setState({
        direction: direction === ASC ? DESC : ASC,
      }, this.historyPush);
    }
  };

  historyPush() {
    const {history} = this.props;
    const {column, direction} = this.state;

    history.push(`/contacts/1/${column}/${direction}`);
  }

  render() {
    const {column, direction} = this.state;

    return (
      <Table.Header>
        <Table.Row>
          <Table.HeaderCell textAlign="center" style={{width: '50px'}}>
            Avatar
          </Table.HeaderCell>
          <Table.HeaderCell
            sorted={column === 'firstName' ? tableDirectionMap[direction] : null}
            onClick={this.handleSort('firstName')}
          >
            First name
          </Table.HeaderCell>
          <Table.HeaderCell
            sorted={column === 'lastName' ? tableDirectionMap[direction] : null}
            onClick={this.handleSort('lastName')}
          >
            Last name
          </Table.HeaderCell>
          <Table.HeaderCell>
            Email
          </Table.HeaderCell>
          <Table.HeaderCell
            sorted={column === 'created' ? tableDirectionMap[direction] : null}
            onClick={this.handleSort('created')}
          >
            Created At
          </Table.HeaderCell>
        </Table.Row>
      </Table.Header>
    );
  }
}

ContactsHeader.propTypes = {
  history: historyType.isRequired,
  match: matchType.isRequired,
};

export default ContactsHeader;
