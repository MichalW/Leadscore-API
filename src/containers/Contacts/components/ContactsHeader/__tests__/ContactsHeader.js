import React from 'react';
import toJson from 'enzyme-to-json';
import {shallow} from 'enzyme';

import ContactsHeader from '../ContactsHeader';
import {DESC} from '../../../../../constants/ActionTypes';

it('should render ContactsHeader', () => {
  const props = {
    history: {
      push: () => {},
    },
    match: {
      params: {
        column: 10,
        direction: DESC,
      },
    },
  };

  const contactsHeader = shallow(<ContactsHeader {...props} />);

  expect(toJson(contactsHeader)).toMatchSnapshot();
});
