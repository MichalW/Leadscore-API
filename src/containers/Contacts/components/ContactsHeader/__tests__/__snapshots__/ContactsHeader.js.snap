// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ContactsHeader 1`] = `
<TableHeader
  as="thead"
>
  <TableRow
    as="tr"
    cellAs="td"
  >
    <TableHeaderCell
      as="th"
      style={
        Object {
          "width": "50px",
        }
      }
      textAlign="center"
    >
      Avatar
    </TableHeaderCell>
    <TableHeaderCell
      as="th"
      onClick={[Function]}
      sorted={null}
    >
      First name
    </TableHeaderCell>
    <TableHeaderCell
      as="th"
      onClick={[Function]}
      sorted={null}
    >
      Last name
    </TableHeaderCell>
    <TableHeaderCell
      as="th"
    >
      Email
    </TableHeaderCell>
    <TableHeaderCell
      as="th"
      onClick={[Function]}
      sorted={null}
    >
      Created At
    </TableHeaderCell>
  </TableRow>
</TableHeader>
`;
